src/NServiceBus.Core.Tests/Persistence/InMemory/When_persisting_a_saga_with_InMemory_and_an_escalated_DTC_transaction.cs

```diff
@@ -0,0 +1,83 @@
+﻿namespace NServiceBus.SagaPersisters.InMemory.Tests
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.Extensibility;
+    using NUnit.Framework;
+    using System.Transactions;
+    using Transports;
+    using Persistence;
+
+    [TestFixture]
+    class When_persisting_a_saga_with_InMemory_and_an_escalated_DTC_transaction
+    {
+        [Test]
+        public async Task Save_fails_when_data_changes_between_concurrent_instances()
+        {
+            var saga = new TestSagaData
+            {
+                Id = Guid.NewGuid()
+            };
+
+            var persister = new InMemorySagaPersister();
+            var storageAdapter = new InMemoryTransactionalSynchronizedStorageAdapter();
+            var insertSession = new InMemorySynchronizedStorageSession();
+
+            await persister.Save(saga, SagaMetadataHelper.GetMetadata<TestSaga>(saga), insertSession, new ContextBag());
+            await insertSession.CompleteAsync();
+
+            var expectedException = Assert.Throws<TransactionAbortedException>(async () =>
```

Comment:
@danielmarbach @WilliamBZA I'm pretty sure it ends up calling `Assert.That` under covers, but honestly I don't even remember the original problem in the first place (something to do with not being able to handle async void methods...?)